jena-arq/src/main/java/org/apache/jena/sparql/util/XSDNumUtils.java

```diff
@@ -147,6 +147,18 @@ public static String canonicalDecimalStrNoIntegerDot(BigDecimal bd) {
         return bd.stripTrailingZeros().toPlainString();
     }
 
+    /** Return a canonical decimal with a trailing ".0". */
+    public static BigDecimal canonicalDecimalWithDot(BigDecimal decimal) {
```

Comment:
How does this differ from `canonicalDecimalStrWithDot`?

Reply:
This new method is used on input after parsing the lexical form into a decimal. The Str Version ist used on output, after converting the decimal to string. I think that the Str Version could be removed in favor of Decimal.toPlainString() since the binary representation is now canonicalized on input.